apps/web/src/app/[lang]/profile/page.tsx

```diff
@@ -0,0 +1,110 @@
+'use client';
+
+import {
+  PersonHead,
+  AgreementsSection,
+  ProposalsSection,
+  DiscussionsSection,
+  MemberSpaces,
+  SpaceGroupSlider,
+} from '@hypha-platform/epics';
+import Link from 'next/link';
+import { useRouter } from 'next/navigation';
+import { ChevronLeftIcon } from '@radix-ui/react-icons';
+import { Text } from '@radix-ui/themes';
+import { Container } from '@hypha-platform/ui';
+import { getDhoPathAgreements } from '../dho/[id]/agreements/constants';
+import {
+  Tabs,
+  TabsTrigger,
+  TabsList,
+  TabsContent,
+} from '@hypha-platform/ui/server';
+import { useMe } from '@web/hooks/use-me';
+import { useParams } from 'next/navigation';
+import { Locale } from '@hypha-platform/i18n';
+import { useEffect } from 'react';
+
+export default function Profile() {
+  const { lang } = useParams();
+  const router = useRouter();
+  const { person } = useMe();
+
+  useEffect(() => {
+    if (
+      person &&
+      typeof person === 'object' &&
+      Object.keys(person).length === 0
```

Comment:
This condition is a nightmare. I think we should be more specific. We should not redirect if `isLoading = true` or `isLoading = false && !!profile`. Generally, this is business logic that we should handle differently. We would like to have a similar redirect anytime we sign in and do not have a profile.

Reply:
> This condition is a nightmare. I think we should be more specific. We should not redirect if `isLoading = true` or `isLoading = false && !!profile`. Generally, this is business logic that we should handle differently. We would like to have a similar redirect anytime we sign in and do not have a profile.

I can remove this here and implement a redirect at the authorization level